File: backend/app/schemas/organism.py
from pydantic import BaseModel


class OrganismSearchResult(BaseModel):
    tax_id: int
    scientific_name: str
    common_name: str | None = None
    lineage: str | None = None


class OrganismDetail(BaseModel):
    tax_id: int
    scientific_name: str
    common_name: str | None = None
    lineage: str | None = None
    gc_content: float | None = None


class CodonTableResponse(BaseModel):
    organism_tax_id: int
    source: str
    table: dict[str, dict[str, float]]  # amino_acid -> {codon: frequency}

File: frontend/src/types/organism.ts
export interface OrganismSearchResult {
  tax_id: number;
  scientific_name: string;
  common_name?: string;
  lineage?: string;
}

export interface OrganismDetail {
  tax_id: number;
  scientific_name: string;
  common_name?: string;
  lineage?: string;
  gc_content?: number;
}

export interface CodonTable {
  organism_tax_id: number;
  source: string;
  table: Record<string, Record<string, number>>;
}
